app/src/org/commcare/android/framework/CommCareActivity.java

```diff
@@ -303,9 +304,8 @@ protected int getWakeLockLevel() {
      * occurred while the activity was paused.
      */
     private void syncTaskBlockingWithDialogFragment() {
-        if (dialogId < 0) {

```

Comment:
@phillipm This is the fix -- don't just blindly dismiss upon resuming, because it messes with the natural lifecycle of re-showing a dialog for a task that is in progress after rotation. Instead, only do so if we have explicitly marked that we tried to dismiss the dialog while paused. (Note that I'm actually not sure this is even necessary; I'm not sure what made you add this check to the previous PR, would be curious to know. But assuming it is necessary, this way of doing it seems safer).